test.js

```diff
@@ -146,6 +146,14 @@ test('Function.prototype.apply with the arguments object', t => {
   t.true(foo('[string|number] <object>', 'bar', {}));
 });
 
+test('allows `this` to be the typeConfig string', t => {
+  function foo () {
+    return argsert.apply('<null|object> [number]', arguments);
```

Comment:
Gotcha; I forget, from Isaacs notes `apply` with arguments had good performance? that makes for a clean API if so.